boards/rp-pico/examples/pico_i2c_oled_display_ssd1306.rs

```diff
@@ -0,0 +1,219 @@
+//! # Raspberry Pi Pico (monochome) 128x64 OLED Display with SSD1306 Driver Example
+//!
+//! This example assumes you got an 128x64 OLED Display with an SSD1306 driver
+//! connected to your Raspberry Pi Pico. The +3.3V voltage source of the
+//! Raspberry Pi Pico will be used, and the output pins 21 and 22 of the board
+//! (on the lower right).
+//!
+//! It will demonstrate how to get an I2C device and use it with the ssd1306 crate.
+//! Additionally you can also see how to format a number into a string using
+//! [core::fmt].
+//!
+//! The following diagram will show how things should be connected.
+//! These displays usually can take 3.3V up to 5V.
+//!
+//! ```text
+//!                              VCC   SCL
+//!                   /------------\    /----------\
+//!                   |        GND  \  /  SDA      |
+//!   _|USB|_         |    /-----\  |  |  /--------+--\
+//!  |1  R 40|        |   /    __|__|__|__|___     |  |
+//!  |2  P 39|        |  /    | ____________  |    |  |
+//!  |3    38|- GND --+-/     | |Hello worl|  |    |  |
+//!  |4  P 37|        |       | |Hello Rust|  |    |  |
+//!  |5  I 36|-+3.3V -/       | |counter: 1|  |    |  |
+//!  |6  C   |                | |          |  |    |  |
+//!  |7  O   |                | """"""""""""  |    |  |
+//!  |       |                 """""""""""""""     |  |
+//!  |       |       (SSD1306 128x64 OLED Display) |  |
+//!  .........                                     /  /
+//!  |       |                                    /  /
+//!  |     22|-GP17 I2C0 SCL---------------------/  /
+//!  |20   21|-GP16 I2C0 SDA-----------------------/
+//!   """""""
+//! Symbols:
+//!     - (+) crossing lines, not connected
+//!     - (o) connected lines
+//! ```
+//!
+//! See the `Cargo.toml` file for Copyright and licence details.
+
+#![no_std]
+#![no_main]
+
+// The macro for our start-up function
+use cortex_m_rt::entry;
+
+// Time handling traits:
+use embedded_time::duration::*;
+use embedded_time::rate::Extensions;
+
+// CountDown timer for the counter on the display:
+use embedded_hal::timer::CountDown;
+
+// Ensure we halt the program on panic (if we don't mention this crate it won't
+// be linked)
+use panic_halt as _;
+
+// A shorter alias for the Peripheral Access Crate, which provides low-level
+// register access
+use rp_pico::hal::pac;
+
+// A shorter alias for the Hardware Abstraction Layer, which provides
+// higher-level drivers.
+use rp_pico::hal;
+
+// For in the graphics drawing utilities like the font
+// and the drawing routines:
+use embedded_graphics::{
+    mono_font::{ascii::FONT_9X18_BOLD, MonoTextStyleBuilder},
+    pixelcolor::BinaryColor,
+    prelude::*,
+    text::{Baseline, Text},
+};
+
+// The display driver:
+use ssd1306::{prelude::*, I2CDisplayInterface, Ssd1306};
```

Comment:
I don't think I'd import I2CDisplayInterface here as it's not clear where it comes from just by the name alone. We only refer to it once so calling it ssd1306::I2CDisplayInterface I think is actually more readable.

Reply:
Makes absolutely sense.